python/paddle/fluid/tests/unittests/eager_op_test.py

```diff
@@ -242,15 +250,55 @@ def __set_elem__(tensor, i, e):
         __set_elem__(tensor_to_check, i, x_pos)
         y_pos = get_output()
 
+        if tensor_to_check_dtype in [np.complex64, np.complex128]:
+            if in_place:
+                set_input(scope, op, inputs, place)
+            x_pos_j = origin + 1j * delta
+            __set_elem__(tensor_to_check, i, x_pos_j)
+            y_pos_j = get_output()
+
         if in_place:
             set_input(scope, op, inputs, place)
 
         x_neg = origin - delta
         __set_elem__(tensor_to_check, i, x_neg)
         y_neg = get_output()
 
+        if tensor_to_check_dtype in [np.complex64, np.complex128]:
+            if in_place:
+                set_input(scope, op, inputs, place)
+
+            x_neg_j = origin - 1j * delta
+            __set_elem__(tensor_to_check, i, x_neg_j)
+            y_neg_j = get_output()
+
+        __set_elem__(tensor_to_check, i, origin)
+
+        if tensor_to_check_dtype in [np.complex64, np.complex128]:
+            # always assume real output, because this function has
+            # no input for dl/dy, though it should do. so there di will be zero
+            f_ajoint = np.ones_like(inputs).astype(tensor_to_check_dtype)
+            df_over_dr = (y_pos - y_neg) / delta / 2
+            df_over_di = (y_pos_j - y_neg_j) / delta / 2
+
+            dl_over_du, dl_over_dv = f_ajoint.real, f_ajoint.imag
+
+            du_over_dx, dv_over_dx = df_over_dr.real, df_over_dr.imag
+
+            du_over_dy, dv_over_dy = df_over_di.real, df_over_di.imag
```

Comment:
`x` and `y` represent inputs and outputs, respectively, it is better to use `du_over_dr`, `du_over_di`, `dv_over_dr`, `dv_over_di` ?

Reply:
Here x and y are x, y in l = u(x,y) + iv(x,y)

Reply:
changed to du_over_dr, du_over_di, dv_over_dr, dv_over_di